kong/runloop/balancer/healthcheckers.lua

```diff
@@ -418,7 +418,11 @@ end
 -- the health checker, this parameter is useful to avoid throwing away current
 -- health status.
 function healthcheckers_M.stop_healthcheckers(delay)
-  for _, id in pairs(upstreams.get_all_upstreams()) do
+  local all_upstreams =  upstreams.get_all_upstreams()
+  if all_upstreams == nil then
+    return
+  end
+  for _, id in pairs(all_upstreams) do
```

Comment:
I suppose this should be enough? (need to require empty table implementation)
```suggestion
  for _, id in pairs(upstreams.get_all_upstreams() or EMPTY_T) do
```

Reply:
I think that's the same with my change, using `EMPTY_T` does not have any additional benefits.

Reply:
Do we need to report the error log `err` if `upstreams.get_all_upstreams()` returnes `nil, err`.

I found that `get_all_upstreams()` always returns a table except if getting some error, then it will return `nil, err`

```
-- Implements a simple dictionary with all upstream-ids indexed
-- by their name.
-- @return The upstreams dictionary (a map with upstream names as string keys
-- and upstream entity tables as values), or nil+error
function upstreams_M.get_all_upstreams()
  local upstreams_dict, err = singletons.core_cache:get("balancer:upstreams", opts,
                                                        load_upstreams_dict_into_memory)
  if err then
    return nil, err
  end

  return upstreams_dict or {}
end
```